### PR TITLE
Bump version to 3.1rc4

### PR DESCRIPTION
**Pull request by twiecki**
After merging this we should not merge any other PRs until 3.1 is out.

### CODE REVIEW

**Comment by fonnesbeck** [comment]
There's one more in `docs/source/conf.py`

**Comment by fonnesbeck** [comment]
The CHANGELOG should also be updated.

**Comment by twiecki** [comment]
Ah, we have release notes, should probably remove CHANGELOG.

**Comment by twiecki** [comment]
conf.py is added.

**Comment by fonnesbeck** [comment]
Are we happy with the release notes as they are? One easy way to get a run-down of changes is:

    git log --oneline --no-merges v3.0..HEAD

Maybe scan through that and see if there are additions we want to highlight.

**Comment by junpenglao** [comment]
I found a bug: #2225

**Comment by fonnesbeck** [comment]
I will test it now too.

**Comment by junpenglao** [comment]
Is there anything left to be done for release?

**Comment by aloctavodia** [comment]
Given that #2294 is a bug fix and not a new feature shouldn't be part of the 3.1 release?

**Comment by twiecki** [comment]
Update RELEASE-NOTES if anything changed, otherwise we should go ahead.

**Comment by junpenglao** [comment]
We should wait for the bug fix of #2298

**Comment by ferrine** [comment]
That's done. Is anything else left?

**Comment by ferrine** [comment]
We then again need experimental 3.2 branch for new features with regular rebase on master.

**Comment by junpenglao** [comment]
Also remove the 3.1 branch and clean up some other branch.

**Comment by junpenglao** [comment]
And also #2258 is still broken.

**Comment by ferrine** [comment]
#2290 is broken too

**Comment by twiecki** [comment]
@fonnesbeck I think we're ready.

**Comment by fonnesbeck** [comment]
OK, will get it rolling today.

**Comment by fonnesbeck** [comment]
Testing the notebooks now. Can we replace `bayesian_neural_network_advi.ipynb` with `bayesian_neural_network_opvi-advi.ipynb`? The former is using an outdated API for part of it, otherwise it is complete overlap.

**Comment by twiecki** [comment]
@fonnesbeck Yes, change the NB.

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -32,7 +32,7 @@
 
   - [Added Stein-Variational Gradient Descent as well as Amortized SVGD (experimental).](https://github.com/pymc-devs/pymc3/pull/2183)
 
-  - [Added pm.generator() to easily specify mini-batches.](http://pymc-devs.github.io/pymc3/notebooks/bayesian_neural_network_opvi-advi.html#Minibatch-ADVI)
+  - [Added `Minibatch` to easily specify mini-batches.](http://pymc-devs.github.io/pymc3/notebooks/bayesian_neural_network_opvi-advi.html#Minibatch-ADVI)
```
Comment (ferrine) [inline RELEASE-NOTES.md]: It is not used in this notebook... I think reference to VI intro is more suitable

**Comment by twiecki** [comment]
I rebased this branch against master, make sure to force pull.

**Comment by twiecki** [comment]
I wonder if we should just go to 3.1 immediately. We've been testing this a lot and had several successful RCs already.

**Comment by fonnesbeck** [comment]
Yes, I was just going to use this RC to ensure that it installs and tests pass, after which I will immediately tag 3.1.

**Comment by fonnesbeck** [comment]
Will do it in a separate PR.

**Comment by junpenglao** [comment]
What about the new docs http://docs.pymc.io/ ?

**Comment by twiecki** [comment]
@junpenglao It's functional but all the links still go the old site, so we would need to figure out a redirect before. I think we better leave that for the next release.

**Comment by junpenglao** [comment]
@twiecki Cool. So I guess we are ready?

**Comment by fonnesbeck** [comment]
OK, will get the official 3.1 going then.

**Comment by twiecki** [comment]
@fonnesbeck trying to upload 3.1rc4 to pypi but airport wifi is not cooperating. Let's do that, everyone install on a clean base and run tests and then go for 3.1.

**Comment by twiecki** [comment]
I think so! @fonnesbeck?

**Comment by fonnesbeck** [comment]
Just pushed a 3.1 PR.

**Pull request merged**